wherehows-web/app/components/datasets/compliance/schema-entities.ts

```diff
@@ -1,7 +1,131 @@
+/**
+ * The ComplianceSchemaEntities component allows the user to individually tag fields to annotate the
+ * data contained in the dataset schema, found in the Compliance tab for datasets.
+ */
+export default class ComplianceSchemaEntities extends Component.extend({}) {
+  /**
+   * Passed in flag determining whether or not we are in an editing mode for the schema entities.
+   * @type {boolean}
+   */
+  isEditing: boolean;
+
+  /**
+   * Passed in computed prop, Lists the IComplianceChangeSet / tags without an identifierType value
+   * @type {Array<IComplianceChangeSet>}
+   */
+  unspecifiedTags: Array<IComplianceChangeSet>;
+
+  /**
+   * Passed in action/method from parent, updates the editing mode and corresponding target
+   */
+  toggleEditing: (e: boolean, target?: ComplianceEdit) => void;
+
+  /**
+   * Passed in action from parent, updates showGuidedEditMode state on parent
+   */
+  // Note: [REFACTOR-COMPLIANCE] This passed in function is to maintain status quo while we migrate the
+  // refactor logic and should eventually no longer be required
+  showGuidedEditMode: (b: boolean) => void;
+
+  /**
+   * Passed in action from parent, updates fieldReviewOption state on parent
+   */
+  // Note: [REFACTOR-COMPLIANCE] Same as above showGuidedEditMode
+  fieldReviewChange: (o: { value: TagFilter }) => TagFilter;
+
+  /**
+   * Specifies the filter to be applied on the list of fields shown in the compliance policy table
+   * @type {TagFilter}
+   * @memberof DatasetCompliance
+   */
+  // Note: [REFACTOR-COMPLIANCE] This value will currently be passed in from the parent but should
+  // eventually live only on this component
+  fieldReviewOption!: TagFilter;
+
+  /**
+   * Used in the template to help pass values for the edit target
+   * @type {ComplianceEdit}
+   */
+  ComplianceEdit = ComplianceEdit;
+
+  /**
+   * References the ComplianceFieldIdValue enum, which specifies hardcoded values for field
+   * identifiers
+   * @type {ComplianceFieldIdValue}
+   */
+  ComplianceFieldIdValue = ComplianceFieldIdValue;
+
+  /**
+   * Enum of categories that can be tracked for this component
+   * @type {TrackableEventCategory}
+   */
+  trackableCategory = TrackableEventCategory;
+
+  /**
+   * Map of events that can be tracked
+   * @type {ITrackableEventCategoryEvent}
+   */
+  trackableEvent = trackableEvent;
+
+  /**
+   * Flag indicating the current compliance policy edit-view mode. Guided edit mode allows users
+   * to go through a wizard to edit the schema entities while the other method is direct JSON editing
+   * @type {boolean}
+   * @memberof ComplianceSchemaEntities
+   */
+  showGuidedComplianceEditMode = true;
+
+  /**
+   * A list of ui values and labels for review filter drop-down
+   * @type {Array<{value: TagFilter, label:string}>}
+   * @memberof ComplianceSchemaEntities
+   */
+  fieldReviewOptions: Array<{ value: ComplianceSchemaEntities['fieldReviewOption']; label: string }> = [
```

Comment:
This was me, but can be corrected to `Array<IDropDownOption<DatasetComplianceEntities['fieldReviewOption']>>`